packages/framework/src/hooks/useEnsembleFormatter.ts

```diff
@@ -0,0 +1,87 @@
+import { getPrettyDate } from "./utils/getPrettyDate";
+import {
+  type PrettyDurationOptions,
+  getPrettyDuration,
+} from "./utils/getPrettyDuration";
+import { getPrettyTime } from "./utils/getPrettyTime";
+
+export interface EnsembleDateTime {
+  getDate: () => string;
+  getDateTime: () => string;
+  prettyDateTime: () => string;
+  prettyDate: () => string;
+  getYear: () => number;
+  getMonth: () => number;
+  getDay: () => number;
+  getDayOfWeek: () => number;
+  getHour: () => number;
+  getMinute: () => number;
+  getSecond: () => number;
+  getMilliseconds: () => number;
+}
+
+export interface EnsembleFormatter {
+  now: () => EnsembleDateTime;
+  prettyDateTime: (input: string) => string;
+  prettyDate: (input: string) => string;
+  prettyDuration: (
+    input: string | number,
+    options?: PrettyDurationOptions,
+  ) => string;
+}
+
+export const useEnsembleFormatter = (): EnsembleFormatter => {
```

Comment:
This is not a hook since it doesn't interact with react state at all. Let's move this into a separate date folder along with its utils